tool/tsh/db.go

```diff
@@ -351,6 +381,113 @@ func onDatabaseConnect(cf *CLIConf) error {
 	return nil
 }
 
+// getDatabaseInfo fetches information about the database from tsh profile is DB is active in profile. Otherwise,
+// the ListDatabases endpoint is called.
+func getDatabaseInfo(cf *CLIConf, tc *client.TeleportClient, dbName string) (*tlsca.RouteToDatabase, error) {
+	database, err := pickActiveDatabase(cf)
+	if err == nil {
+		return database, nil
+	}
+	if !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
+	db, err := getDatabaseResourceByName(cf, tc, dbName)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &tlsca.RouteToDatabase{
+		ServiceName: db.GetName(),
+		Protocol:    db.GetProtocol(),
+		Username:    cf.Username,
+		Database:    cf.DatabaseName,
+	}, nil
+}
+
+func getDatabaseResourceByName(cf *CLIConf, tc *client.TeleportClient, dbName string) (types.Database, error) {
+	var databases []types.Database
+	err := client.RetryWithRelogin(cf.Context, tc, func() error {
+		allDatabases, err := tc.ListDatabases(cf.Context)
+		for _, database := range allDatabases {
+			if database.GetName() == dbName {
+				databases = append(databases, database)
+			}
+		}
+		return trace.Wrap(err)
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if len(databases) == 0 {
+		return nil, trace.NotFound(
+			"database %q not found, use 'tsh db ls' to see registered databases", dbName)
+	}
+	return databases[0], nil
+}
+
+func dbCertExpiredOrMFAIsRequired(cf *CLIConf, tc *client.TeleportClient, database *tlsca.RouteToDatabase, profile *client.ProfileStatus) (bool, error) {
+	// Load DB cert and check if cert is still valid. Missing cert is treated as expired cert.
+	certExpired, err := isExpiredCert(profile.DatabaseCertPath(database.ServiceName))
```

Comment:
ProfileStatus has `IsExpired` method. Can it be used instead? If it can't be used for database access (e.g. if db session is shorter), let's implement this expiration check under ProfileStatus as well similar to ValidUntil/IsExpired it has for "main" identity.

Reply:
> ProfileStatus has IsExpired method. Can it be used instead?

I don't think so, profile.IsExpired uses `sshCert` expiration time but in this case we would like to check particular db cert expiration time.

> let's implement this expiration check under ProfileStatus as well similar to ValidUntil/IsExpired it has for "main" identity.

Right now user profile stores only `tlsca.RouteToDatabase` information but I can extend that by something like: 
```diff
type ProfileStatus struct {
-       tlsca.RouteToDatabase
+       Databases []DatabaseStatus
}

+type DatabaseStatus struct {
+	tlsca.RouteToDatabase
+	ValidUntil time.Time
+}
```
so `profile.Databases[..].IsExpired()` can be called instead `isExpiredCert(profile.DatabaseCertPath(database.ServiceName))`